packages/contracts/test/utils/combinatorial_sets.ts

```diff
@@ -0,0 +1,25 @@
+import { BigNumber } from '@0xproject/utils';
+import * as _ from 'lodash';
+
+export const BigNumbers = [];
+
+const MAX_256_BIT_NUMBER = new BigNumber(2).pow(256).minus(1);
+
+export const positiveNaturalBigNumbers = [
+    new BigNumber(0),
+    new BigNumber(1),
+    new BigNumber(2),
+    new BigNumber(2).pow(64),
+    MAX_256_BIT_NUMBER,
+    MAX_256_BIT_NUMBER.div(2).floor(),
+    MAX_256_BIT_NUMBER.sqrt().floor(),
```

Comment:
This evaluated to `2**128 - 1`, which does *not* cause overflow when multiplied by itself.

I would avoid using the `div` and `sqrt` function and instead specify the numbers as directly as possible (either as `new BigNumber(2).pow(255).sub(1)` or as `0x7FFFFFFFFFFFFFF...`.

As it stands it is not clear if you intended to be exactly on or just below the threshold of overflow. A comment can also clarify this.

Reply:
👍I'm going to update this to use both the *maximum* value that *should not* overflow and the *minimum* value that *should* overflow under certain conditions. We can then use a reference function to differentiate between cases where we expect or do not expect an overflow.